code/modules/clothing/head/hardhat.dm

```diff
@@ -37,6 +37,13 @@
 	update_action_button_icons()
 	update_icon()
 
+/obj/item/clothing/head/hardhat/attack_alien(mob/living/carbon/xenomorph/X, isrightclick = FALSE)
+	if(turn_light(X, FALSE) != CHECKS_PASSED)
```

Comment:
this should call parent, theres a signal

Reply:
Are you sure? I copy pasta this from flashlights, so should I fix them too?

https://github.com/tgstation/TerraGov-Marine-Corps/blob/587e7a5a40d78a0eaf15a5f663afd8755891f15a/code/game/objects/items.dm#L940-L943